File: Tree/94.java
// 94. Binary Tree Inorder Traversal

// 11-26

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
package Tree;
import java.util.ArrayList;
import java.util.List;
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> list = new ArrayList<Integer>();
        helper(root, list);
        return list;
    }

    public void helper(TreeNode node, List<Integer> list) {
        if (node == null)
            return;
        helper(node.left, list);
        list.add(node.val);
        helper(node.right, list);
    }
}

File: Tree/94.py
# 94. Binary Tree Inorder Traversal
# 11-26


# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        list = []
        inorderHelper(root, list)
        return list


def inorderHelper(node, list):
    if node is None:
        return
    inorderHelper(node.left, list)
    list.append(node.val)
    inorderHelper(node.right, list)
